solr/webapp/web/solrUi/src/app/modules/common/utils/DateConverter.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class DateConverter {
+  private static pattern: string = '/.* (?<date>\d{4}-\d{2}-\d{2}) .*/';
```

Comment:
Are you telling me that the line that I receive from API for solr impl version is going to change it format?
`9.7.0 675a41516e3f3bacfc975590773e7abdca444ff43e7abdca444ff4 - anshum - 2024-09-03 15:05:20`
Can you provide me what other formats it might have?

Reply:
I am not sure how it is generated, but it may change depending on the configured locale of the JVM / Solr instance. And I believe this place is not the only place you want to use a date converter.

In the logging section for example dates can also switch format (at least in the frontend). There, you may get other formats from the API if it is not standardized.

But my comment is more about "if there is already a solution from the framework, use that instead", like the [DatePipe](https://angular.dev/api/common/DatePipe) in this situation, so you don't have to worry about maintenance and changes. :)

Reply:
Well, if we're going to discuss the current value then, my solution is for working with that part, it cuts DateTime value out of string value.
Why I'm not using default pipe just because the string that I'm getting is non DateTime only, it also holds the data that has nothing to do with that. In other cases, I would use default pipe transformation rather creating my own solution.

Reply:
Makes sense and I agree with you here. We may consider not touching this value though, as it is part of a "version code" that may be used in bug reporting (date localization would not make much sense I think). But that is only a minor opinionated "improvement". :)